internal/proofs/convert.go

```diff
@@ -0,0 +1,98 @@
+package proofs
```

Comment:
why do we have an internal pkg?

Reply:
these types and functions don't need to be exposed. I can remove it but was trying to avoid expanding the surface of exposed things